### PR TITLE
Extend REST API with option to add and update notes

### PR DESCRIPTION
**Pull request by J535D165**
#832 and #820. 

It requires also changes to the javascript code @terrymyc. Can you add that to this PR?

@peterlombaers, Can you review the "update_decision" method? Made this one slightly more flexible because everything is now properly handled in the javascript code (the difference between PUT and POST)

### CODE REVIEW

File: asreview/state/sqlstate.py [hunk 1]
```diff
@@ -620,24 +620,19 @@ def add_labeling_data(self, record_ids, labels, notes=None, prior=False):
...
         # Change the label.
-        cur.execute("UPDATE results SET label = ? WHERE record_id = ?",
-                    (new_label, record_id))
+        cur.execute("UPDATE results SET label = ?, notes = ? WHERE record_id = ?",
+                    (label, note, record_id))
 
         # Add the change to the decision changes table.
         cur.execute("INSERT INTO decision_changes VALUES (?,?, ?)",
-                    (record_id, new_label, current_time))
+                    (record_id, label, datetime.now()))
```
Comment (PeterLombaers) [inline asreview/state/sqlstate.py:635]: Do we want to update this table if we only add a note, but keep the label the same? Or we could add the notes to the decision_changes table, that way you also save the history of the notes. Note sure if that is desirable though.

**Review by PeterLombaers — COMMENTED** [review]
Just one comment regarding the decision_changes table, other than that it looks good. I like it better this way, it is more flexible indeed.

File: asreview/state/sqlstate.py (hunk 1 above)
Comment (J535D165) [reply asreview/state/sqlstate.py:635]: I was thinking about privacy-related issues. A decision doesn't harm anyone. However, a note can contain info like "Discuss with John Doe". The note is removed by the reviewer after the discussion, however, it's never removed from the history.

Comment (PeterLombaers) [reply asreview/state/sqlstate.py:635]: Yes, that makes sense. I agree that you should be able to remove notes, but not the previous labels.

**Review by terrymyc — APPROVED** [review]
I think I need to start another PR to implement the note functionality in the front end after this is merged.

**Pull request merged**